lib/puppet-syntax/templates.rb

```diff
@@ -29,11 +29,33 @@ def check(filelist)
     def validate_epp(filename)
       require 'puppet/pops'
       errors = []
+      evaluating_parser = Puppet::Pops::Parser::EvaluatingParser::EvaluatingEppParser.new()
+      parser = evaluating_parser.parser()
       begin
-        parser = Puppet::Pops::Parser::EvaluatingParser::EvaluatingEppParser.new()
-        parser.parse_file(filename)
-      rescue => detail
-        errors << detail
+        parse_result = parser.parse_file(filename)
+        validation_result = evaluating_parser.validate(parse_result.model)
+
+        # print out any warnings
+        validation_result.warnings.each do |warn|
+          formatter = Puppet::Pops::Validation::DiagnosticFormatterPuppetStyle.new
+          message = formatter.format_message(warn)
+          file = warn.file
+          line = warn.source_pos.line
+          column = warn.source_pos.pos
+          STDOUT.puts("WARNING: #{file}:#{line}:#{column}: #{message}")
```

Comment:
It would be nice if this method didn't do anything with stdout. Perhaps it should return an `OpenStruct` with an array of warnings and an array of errors.

Reply:
The template checks are now returning a ruby `Hash` with `:warnings` and `:errors` keys, each being an `Array`.
The `STDOUT`/`STDERR` output is now only done in the rake task again.